File: lib_core/src/main/java/com/eelly/core/widget/refresh/listener/OnMultiPurposeListener.java
package com.eelly.core.widget.refresh.listener;


import com.eelly.core.widget.refresh.api.IRefreshFooter;
import com.eelly.core.widget.refresh.api.IRefreshHeader;

/**
 * @author Vurtne on 14-Dec-17.
 */

public interface OnMultiPurposeListener extends OnRefreshLoadMoreListener, OnStateChangedListener {

    void onHeaderPulling(IRefreshHeader header, float percent, int offset, int headerHeight, int extendHeight);
    void onHeaderReleased(IRefreshHeader header, int headerHeight, int extendHeight);
    void onHeaderReleasing(IRefreshHeader header, float percent, int offset, int headerHeight, int extendHeight);
    void onHeaderStartAnimator(IRefreshHeader header, int headerHeight, int extendHeight);
    void onHeaderFinish(IRefreshHeader header, boolean success);

    void onFooterPulling(IRefreshFooter footer, float percent, int offset, int footerHeight, int extendHeight);
    void onFooterReleased(IRefreshFooter footer, int footerHeight, int extendHeight);
    void onFooterReleasing(IRefreshFooter footer, float percent, int offset, int footerHeight, int extendHeight);
    void onFooterStartAnimator(IRefreshFooter footer, int footerHeight, int extendHeight);
    void onFooterFinish(IRefreshFooter footer, boolean success);
}

File: lib_core/src/main/java/com/eelly/core/widget/refresh/listener/OnMultiPurposeListener.kt
package com.eelly.core.widget.refresh.listener

import com.eelly.core.widget.refresh.api.IRefreshFooter
import com.eelly.core.widget.refresh.api.IRefreshHeader

/**
 * @author Vurtne on 14-Dec-17.
 */
interface OnMultiPurposeListener {
    fun onHeaderPulling(header: IRefreshHeader, percent: Float, offset: Int, headerHeight: Int, extendHeight: Int)
    fun onHeaderReleased(header: IRefreshHeader, headerHeight: Int, extendHeight: Int)
    fun onHeaderReleasing(header: IRefreshHeader, percent: Float, offset: Int, headerHeight: Int, extendHeight: Int)
    fun onHeaderStartAnimator(header: IRefreshHeader, headerHeight: Int, extendHeight: Int)
    fun onHeaderFinish(header: IRefreshHeader, success: Boolean)

    fun onFooterPulling(footer: IRefreshFooter, percent: Float, offset: Int, footerHeight: Int, extendHeight: Int)
    fun onFooterReleased(footer: IRefreshFooter, footerHeight: Int, extendHeight: Int)
    fun onFooterReleasing(footer: IRefreshFooter, percent: Float, offset: Int, footerHeight: Int, extendHeight: Int)
    fun onFooterStartAnimator(footer: IRefreshFooter, footerHeight: Int, extendHeight: Int)
    fun onFooterFinish(footer: IRefreshFooter, success: Boolean)
}